pkg/kubelet/kubelet.go

```diff
@@ -2786,3 +2786,17 @@ func extractBandwidthResources(pod *api.Pod) (ingress, egress *resource.Quantity
 	}
 	return ingress, egress, nil
 }
+
+// Runs the command in the container
+func (kl *Kubelet) RunContainerCommand(pod *api.Pod, container *api.Container, cmd []string) ([]byte, error) {
+	// Mount volumes.
+	podFullName := kubecontainer.GetPodFullName(pod)
+	podVolumes, err := kl.mountExternalVolumes(pod)
```

Comment:
kubelet builds volume first then creates the pod. if there is no volume, kubelet will [complain](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet.go#L959).